src/System.Net.Security/tests/FunctionalTests/KerberosTest.cs

```diff
@@ -0,0 +1,331 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net.Test.Common;
+using System.Security.Authentication;
+using System.Security.Principal;
+using System.Text;
+using System.Threading.Tasks;
+
+using Xunit;
+
+namespace System.Net.Security.Tests
+{
+    public class KDCSetup
+    {
+        private const string Krb5ConfigFile = "/etc/krb5.conf";
+        private const string KDestroyCmd = "kdestroy";
+
+        public KDCSetup()
+        {
+            ProcessStartInfo startInfo = new ProcessStartInfo();
+            startInfo.UseShellExecute = true;
+            startInfo.FileName = "./setup-kdc.sh";
```

Comment:
We don't want to configure it on jenkins, as the jenkins machines are shared over all dotnet jobs across various repos. And since this setup is very specific to negotiatestream it made since to have it here. Another reason is these tests will surely fail on all non-CI machines (local dev boxes) while running outerloop if we move the script as a jenkins setup.